plugins/pastefromword/plugin.js

```diff
@@ -96,7 +96,18 @@
 						editor.fire( 'paste', data );
 					} else if ( !editor.config.pasteFromWordPromptCleanup || ( forceFromWord || confirm( editor.lang.pastefromword.confirmCleanup ) ) ) {
 
-						pfwEvtData.dataValue = CKEDITOR.cleanWord( pfwEvtData.dataValue, editor );
+						var filteredData = CKEDITOR.cleanWord( pfwEvtData.dataValue, editor );
+
+						if ( typeof filteredData === 'string' ) {
+							pfwEvtData.dataValue = filteredData;
+						} else if ( typeof filteredData === 'object' ) {
+							editor.fire( 'lockSnapshot' );
+							pfwEvtData.dataValue = filteredData.dataValue;
+							handleBlobs( filteredData.blobUrls, function() {
```

Comment:
I'm wondering about time between `lockSnapshot` and `unlockSnapshot`. If user could write anything in this time, it could result in broken or at least weird snapshot.

Reply:
@Comandeer, yes it's a drawback of this approach. Another option is to make editor somehow readOnly, but I'm not sure what would be better here. WDYT? Should editor be read only during this paste? Or maybe you have some other solutions which I don't see?

Reply:
Making editor read-only sounds even worse. Let's leave it for now as it shouldn't be so critical issue.